### ISSUE TITLE
visualize() not outputting anything for Delayed(xarray object)

### ISSUE THREAD

**Issue by TomNicholas** [dask/dask#5721]
Hi, I'm trying to visualise the graph involved when xarray writes to a netcdf file, but no visualisation is appearing.

```python
import dask
import xarray

ds = xarray.Dataset({'a': 0})
task = ds.to_netcdf('dat.nc',compute=False)
task.compute()
print(task)
task.visualise(filename='netcdf.svg')  # This file is never produced


def inc(x):
    return x + 1

z = dask.delayed(inc)(1)
z.compute()
print(z)
z.visualize(filename='dask_graph.svg')  # This gets written fine
```

The `.to_netcdf()` task is not producing any image or file output, while the example from dask's documentation is. That's very confusing because in both cases the `.visualize()` method is being called on a `Delayed()` object - the console printout is:

```
Delayed('_finalize_store-00c1f2a6-b61e-4d3f-9b17-e4d5bd01cf1c')
Delayed('inc-6224bc6f-721e-4f8c-8dda-89634800cd2e')
```

Do I have to install something else?

<details>
<summary>
Package versions used
</summary>
<p>

```bash
$ conda list
```
```
_ipyw_jlab_nb_ext_conf    0.1.0                    py37_0  
_libgcc_mutex             0.1                        main  
alabaster                 0.7.12                   py37_0  
anaconda                  2019.10                  py37_0  
anaconda-client           1.7.2                    py37_0  
anaconda-navigator        1.9.7                    py37_0  
anaconda-project          0.8.3                      py_0  
animatplot                0.4.1                    pypi_0    pypi
asn1crypto                1.0.1                    py37_0  
astroid                   2.3.1                    py37_0  
astropy                   3.2.2            py37h7b6447c_0  
atomicwrites              1.3.0                    py37_1  
attrs                     19.2.0                     py_0  
babel                     2.7.0                      py_0  
backcall                  0.1.0                    py37_0  
backports                 1.0                        py_2  
backports.functools_lru_cache 1.6.1                      py_0  
backports.os              0.1.1                    py37_0  
backports.shutil_get_terminal_size 1.0.0                    py37_2  
backports.tempfile        1.0                        py_1  
backports.weakref         1.0.post1                  py_1  
beautifulsoup4            4.8.0                    py37_0  
bitarray                  1.0.1            py37h7b6447c_0  
bkcharts                  0.2                      py37_0  
blas                      1.0                         mkl  
bleach                    3.1.0                    py37_0  
blosc                     1.16.3               hd408876_0  
bokeh                     1.3.4                    py37_0  
boto                      2.49.0                   py37_0  
bottleneck                1.2.1            py37h035aef0_1  
bzip2                     1.0.8                h7b6447c_0  
ca-certificates           2019.8.28                     0  
cairo                     1.14.12              h8948797_3  
certifi                   2019.9.11                py37_0  
cffi                      1.12.3           py37h2e261b9_0  
cftime                    1.0.4.2                  pypi_0    pypi
chardet                   3.0.4                 py37_1003  
click                     7.0                      py37_0  
cloudpickle               1.2.2                      py_0  
clyent                    1.2.2                    py37_1  
colorama                  0.4.1                    py37_0  
conda-package-handling    1.6.0            py37h7b6447c_0  
conda-verify              3.4.2                      py_1  
configobj                 5.0.6                    pypi_0    pypi
contextlib2               0.6.0                      py_0  
cryptography              2.7              py37h1ba5d50_0  
curl                      7.65.3               hbc83047_0  
cycler                    0.10.0                   py37_0  
cython                    0.29.13          py37he6710b0_0  
cytoolz                   0.10.0           py37h7b6447c_0  
dask                      2.5.2                      py_0  
dask-core                 2.5.2                      py_0  
dbus                      1.13.6               h746ee38_0  
decorator                 4.4.0                    py37_1  
defusedxml                0.6.0                      py_0  
distributed               2.5.2                      py_0  
docutils                  0.15.2                   py37_0  
entrypoints               0.3                      py37_0  
et_xmlfile                1.0.1                    py37_0  
expat                     2.2.6                he6710b0_0  
fastcache                 1.1.0            py37h7b6447c_0  
filelock                  3.0.12                     py_0  
flask                     1.1.1                      py_0  
fontconfig                2.13.0               h9420a91_0  
freetype                  2.9.1                h8a8886c_1  
fribidi                   1.0.5                h7b6447c_0  
fsspec                    0.5.2                      py_0  
future                    0.18.2                   py37_0  
get_terminal_size         1.0.0                haa9412d_0  
gevent                    1.4.0            py37h7b6447c_0  
glib                      2.56.2               hd408876_0  
glob2                     0.7                        py_0  
gmp                       6.1.2                h6c8ec71_1  
gmpy2                     2.0.8            py37h10f8cd9_2  
graphite2                 1.3.13               h23475e2_0  
graphviz                  2.40.1               h21bd128_2  
greenlet                  0.4.15           py37h7b6447c_0  
gst-plugins-base          1.14.0               hbbd80ab_1  
gstreamer                 1.14.0               hb453b48_1  
h5py                      2.9.0            py37h7918eee_0  
harfbuzz                  1.8.8                hffaf4a1_0  
hdf5                      1.10.4               hb1b8bf9_0  
heapdict                  1.0.1                      py_0  
html5lib                  1.0.1                    py37_0  
icu                       58.2                 h9c2bf20_1  
idna                      2.8                      py37_0  
imageio                   2.6.0                    py37_0  
imagesize                 1.1.0                    py37_0  
importlib_metadata        0.23                     py37_0  
intel-openmp              2019.4                      243  
ipykernel                 5.1.2            py37h39e3cac_0  
ipython                   7.8.0            py37h39e3cac_0  
ipython_genutils          0.2.0                    py37_0  
ipywidgets                7.5.1                      py_0  
isort                     4.3.21                   py37_0  
itsdangerous              1.1.0                    py37_0  
jbig                      2.1                  hdba287a_0  
jdcal                     1.4.1                      py_0  
jedi                      0.15.1                   py37_0  
jeepney                   0.4.1                      py_0  
jinja2                    2.10.3                     py_0  
joblib                    0.13.2                   py37_0  
jpeg                      9b                   h024ee3a_2  
json5                     0.8.5                      py_0  
jsonschema                3.0.2                    py37_0  
jupyter                   1.0.0                    py37_7  
jupyter_client            5.3.3                    py37_1  
jupyter_console           6.0.0                    py37_0  
jupyter_core              4.5.0                      py_0  
jupyterlab                1.1.4              pyhf63ae98_0  
jupyterlab_server         1.0.6                      py_0  
keyring                   18.0.0                   py37_0  
kiwisolver                1.1.0            py37he6710b0_0  
krb5                      1.16.1               h173b8e3_7  
lazy-object-proxy         1.4.2            py37h7b6447c_0  
libarchive                3.3.3                h5d8350f_5  
libcurl                   7.65.3               h20c2e04_0  
libedit                   3.1.20181209         hc058e9b_0  
libffi                    3.2.1                hd88cf55_4  
libgcc-ng                 9.1.0                hdf63c60_0  
libgfortran-ng            7.3.0                hdf63c60_0  
liblief                   0.9.0                h7725739_2  
libpng                    1.6.37               hbc83047_0  
libsodium                 1.0.16               h1bed415_0  
libssh2                   1.8.2                h1ba5d50_0  
libstdcxx-ng              9.1.0                hdf63c60_0  
libtiff                   4.0.10               h2733197_2  
libtool                   2.4.6                h7b6447c_5  
libuuid                   1.0.3                h1bed415_2  
libxcb                    1.13                 h1bed415_1  
libxml2                   2.9.9                hea5a465_1  
libxslt                   1.1.33               h7d1a2b0_0  
llvmlite                  0.29.0           py37hd408876_0  
locket                    0.2.0                    py37_1  
lxml                      4.4.1            py37hefd8a0e_0  
lz4-c                     1.8.1.2              h14c3975_0  
lzo                       2.10                 h49e0be7_2  
markupsafe                1.1.1            py37h7b6447c_0  
matplotlib                3.1.1            py37h5429711_0  
mccabe                    0.6.1                    py37_1  
mistune                   0.8.4            py37h7b6447c_0  
mkl                       2019.4                      243  
mkl-service               2.3.0            py37he904b0f_0  
mkl_fft                   1.0.14           py37ha843d7b_0  
mkl_random                1.1.0            py37hd6b4f25_0  
mock                      3.0.5                    py37_0  
more-itertools            7.2.0                    py37_0  
mpc                       1.1.0                h10f8cd9_1  
mpfr                      4.0.1                hdf1c602_3  
mpmath                    1.1.0                    py37_0  
msgpack-python            0.6.1            py37hfd86e86_1  
multipledispatch          0.6.0                    py37_0  
natsort                   6.2.0                    pypi_0    pypi
navigator-updater         0.2.1                    py37_0  
nbconvert                 5.6.0                    py37_1  
nbformat                  4.4.0                    py37_0  
ncurses                   6.1                  he6710b0_1  
netcdf4                   1.5.3                    pypi_0    pypi
networkx                  2.3                        py_0  
nltk                      3.4.5                    py37_0  
nose                      1.3.7                    py37_2  
notebook                  6.0.1                    py37_0  
numba                     0.45.1           py37h962f231_0  
numexpr                   2.7.0            py37h9e4a6bb_0  
numpy                     1.17.2           py37haad9e8e_0  
numpy-base                1.17.2           py37hde5b4d6_0  
numpydoc                  0.9.1                      py_0  
olefile                   0.46                     py37_0  
openpyxl                  3.0.0                      py_0  
openssl                   1.1.1d               h7b6447c_2  
packaging                 19.2                       py_0  
pandas                    0.25.1           py37he6710b0_0  
pandoc                    2.2.3.2                       0  
pandocfilters             1.4.2                    py37_1  
pango                     1.42.4               h049681c_0  
parso                     0.5.1                      py_0  
partd                     1.0.0                      py_0  
patchelf                  0.9                  he6710b0_3  
path.py                   12.0.1                     py_0  
pathlib2                  2.3.5                    py37_0  
patsy                     0.5.1                    py37_0  
pcre                      8.43                 he6710b0_0  
pep8                      1.7.1                    py37_0  
pexpect                   4.7.0                    py37_0  
pickleshare               0.7.5                    py37_0  
pillow                    6.2.0            py37h34e0f95_0  
pip                       19.2.3                   py37_0  
pixman                    0.38.0               h7b6447c_0  
pkginfo                   1.5.0.1                  py37_0  
pluggy                    0.13.0                   py37_0  
ply                       3.11                     py37_0  
prometheus_client         0.7.1                      py_0  
prompt_toolkit            2.0.10                     py_0  
psutil                    5.6.3            py37h7b6447c_0  
ptyprocess                0.6.0                    py37_0  
py                        1.8.0                    py37_0  
py-lief                   0.9.0            py37h7725739_2  
pycodestyle               2.5.0                    py37_0  
pycosat                   0.6.3            py37h14c3975_0  
pycparser                 2.19                     py37_0  
pycrypto                  2.6.1            py37h14c3975_9  
pycurl                    7.43.0.3         py37h1ba5d50_0  
pyflakes                  2.1.1                    py37_0  
pygments                  2.4.2                      py_0  
pylint                    2.4.2                    py37_0  
pyodbc                    4.0.27           py37he6710b0_0  
pyopenssl                 19.0.0                   py37_0  
pyparsing                 2.4.2                      py_0  
pyqt                      5.9.2            py37h05f1152_2  
pyrsistent                0.15.4           py37h7b6447c_0  
pysocks                   1.7.1                    py37_0  
pytables                  3.5.2            py37h71ec239_1  
pytest                    5.2.1                    py37_0  
pytest-arraydiff          0.3              py37h39e3cac_0  
pytest-astropy            0.5.0                    py37_0  
pytest-doctestplus        0.4.0                      py_0  
pytest-openfiles          0.4.0                      py_0  
pytest-remotedata         0.3.2                    py37_0  
python                    3.7.4                h265db76_1  
python-dateutil           2.8.0                    py37_0  
python-graphviz           0.13.2                     py_0    conda-forge
python-libarchive-c       2.8                     py37_13  
pytz                      2019.3                     py_0  
pywavelets                1.0.3            py37hdd07704_1  
pyyaml                    5.1.2            py37h7b6447c_0  
pyzmq                     18.1.0           py37he6710b0_0  
qt                        5.9.7                h5867ecd_1  
qtawesome                 0.6.0                      py_0  
qtconsole                 4.5.5                      py_0  
qtpy                      1.9.0                      py_0  
readline                  7.0                  h7b6447c_5  
requests                  2.22.0                   py37_0  
ripgrep                   0.10.0               hc07d326_0  
rope                      0.14.0                     py_0  
ruamel_yaml               0.15.46          py37h14c3975_0  
scikit-image              0.15.0           py37he6710b0_0  
scikit-learn              0.21.3           py37hd81dba3_0  
scipy                     1.3.1            py37h7c811a0_0  
seaborn                   0.9.0                    py37_0  
secretstorage             3.1.1                    py37_0  
send2trash                1.5.0                    py37_0  
setuptools                41.4.0                   py37_0  
simplegeneric             0.8.1                    py37_2  
singledispatch            3.4.0.3                  py37_0  
sip                       4.19.8           py37hf484d3e_0  
six                       1.12.0                   py37_0  
snappy                    1.1.7                hbae5bb6_3  
snowballstemmer           2.0.0                      py_0  
sortedcollections         1.1.2                    py37_0  
sortedcontainers          2.1.0                    py37_0  
soupsieve                 1.9.3                    py37_0  
sphinx                    2.2.0                      py_0  
sphinxcontrib             1.0                      py37_1  
sphinxcontrib-applehelp   1.0.1                      py_0  
sphinxcontrib-devhelp     1.0.1                      py_0  
sphinxcontrib-htmlhelp    1.0.2                      py_0  
sphinxcontrib-jsmath      1.0.1                      py_0  
sphinxcontrib-qthelp      1.0.2                      py_0  
sphinxcontrib-serializinghtml 1.1.3                      py_0  
sphinxcontrib-websupport  1.1.2                      py_0  
spyder                    3.3.6                    py37_0  
spyder-kernels            0.5.2                    py37_0  
sqlalchemy                1.3.9            py37h7b6447c_0  
sqlite                    3.30.0               h7b6447c_0  
statsmodels               0.10.1           py37hdd07704_0  
sympy                     1.4                      py37_0  
tbb                       2019.4               hfd86e86_0  
tblib                     1.4.0                      py_0  
terminado                 0.8.2                    py37_0  
testpath                  0.4.2                    py37_0  
tk                        8.6.8                hbc83047_0  
toolz                     0.10.0                     py_0  
tornado                   6.0.3            py37h7b6447c_0  
tqdm                      4.36.1                     py_0  
traitlets                 4.3.3                    py37_0  
unicodecsv                0.14.1                   py37_0  
unixodbc                  2.3.7                h14c3975_0  
urllib3                   1.24.2                   py37_0  
wcwidth                   0.1.7                    py37_0  
webencodings              0.5.1                    py37_1  
werkzeug                  0.16.0                     py_0  
wheel                     0.33.6                   py37_0  
widgetsnbextension        3.5.1                    py37_0  
wrapt                     1.11.2           py37h7b6447c_0  
wurlitzer                 1.0.3                    py37_0  
xarray                    0.14.1+28.gf2b2f9f           dev_0    <develop>
xbout                     0.1.0                     dev_0    <develop>
xlrd                      1.2.0                    py37_0  
xlsxwriter                1.2.1                      py_0  
xlwt                      1.3.0                    py37_0  
xz                        5.2.4                h14c3975_4  
yaml                      0.1.7                had09818_2  
zeromq                    4.3.1                he6710b0_3  
zict                      1.0.0                      py_0  
zipp                      0.6.0                      py_0  
zlib                      1.2.11               h7b6447c_3  
zstd                      1.3.7                h0b5b093_0
```
</p>
</details>

**Comment by TomAugspurger** [comment]
The method is visuali**z**e, with a `z` :) You're the second person to run into this.

It may be worth aliasing `visualise` to `visualize`, given that you don't get an AttributeError with dask.delayed.

**Comment by TomNicholas** [comment]
> visuali**z**e, with a `z`

Oh wow :sweat_smile:... Yeah that fixed it haha

> you don't get an AttributeError with dask.delayed

That seems weird, why not? Did I not just call a method which isn't defined?

> It may be worth aliasing visualise to visualize

I'm happy to submit a quick PR for this?

**Comment by jrbourbeau** [comment]
> I'm happy to submit a quick PR for this?

Thanks would be very welcome, thanks @TomNicholas! 

> That seems weird, why not? Did I not just call a method which isn't defined?

Since `delayed` results are lazily evaluated, `AttributeError`s don't arise until an actual computation takes place. For example:

```python
In [1]: from dask import delayed

In [2]: class Foo:
   ...:     pass
   ...:

In [3]: f = Foo()

In [4]: f_delayed = delayed(f)

In [5]: result = f_delayed.bar()   # lazily evaluated

In [6]: result
Out[6]: Delayed('bar-561758a9-4bc2-4baa-84b6-786bfb8d667a')

In [7]: result.compute()    # will perform computation and raise an AttributeError
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-7-a4f886589eb2> in <module>
----> 1 result.compute()

~/github/dask/dask/dask/base.py in compute(self, **kwargs)
    163         dask.base.compute
    164         """
--> 165         (result,) = compute(self, traverse=False, **kwargs)
    166         return result
    167

~/github/dask/dask/dask/base.py in compute(*args, **kwargs)
    434     keys = [x.__dask_keys__() for x in collections]
    435     postcomputes = [x.__dask_postcompute__() for x in collections]
--> 436     results = schedule(dsk, keys, **kwargs)
    437     return repack([f(r, *a) for r, (f, a) in zip(results, postcomputes)])
    438

~/github/dask/dask/dask/threaded.py in get(dsk, result, cache, num_workers, pool, **kwargs)
     79         get_id=_thread_get_id,
     80         pack_exception=pack_exception,
---> 81         **kwargs
     82     )
     83

~/github/dask/dask/dask/local.py in get_async(apply_async, num_workers, dsk, result, cache, get_id, rerun_exceptions_locally, pack_exception, raise_exception, callbacks, dumps, loads, **kwargs)
    484                         _execute_task(task, data)  # Re-execute locally
    485                     else:
--> 486                         raise_exception(exc, tb)
    487                 res, worker_id = loads(res_info)
    488                 state["cache"][key] = res

~/github/dask/dask/dask/local.py in reraise(exc, tb)
    314     if exc.__traceback__ is not tb:
    315         raise exc.with_traceback(tb)
--> 316     raise exc
    317
    318

~/github/dask/dask/dask/local.py in execute_task(key, task_info, dumps, loads, get_id, pack_exception)
    220     try:
    221         task, data = loads(task_info)
--> 222         result = _execute_task(task, data)
    223         id = get_id()
    224         result = dumps((result, id))

~/github/dask/dask/dask/core.py in _execute_task(arg, cache, dsk)
    117         func, args = arg[0], arg[1:]
    118         args2 = [_execute_task(a, cache) for a in args]
--> 119         return func(*args2)
    120     elif not ishashable(arg):
    121         return arg

~/github/dask/dask/dask/utils.py in __call__(self, obj, *args, **kwargs)
    879
    880     def __call__(self, obj, *args, **kwargs):
--> 881         return getattr(obj, self.method)(*args, **kwargs)
    882
    883     def __reduce__(self):

AttributeError: 'Foo' object has no attribute 'bar'
```

**Comment by quasiben** [comment]
@TomNicholas are you still interested in submitting a PR for this?  If not, no worries

**Comment by 1cybersheep1** [comment]
I'm interested on doing this. Should i just create a method called visualise to alias visualize?

**Comment by mindhash** [comment]
@quasiben I added alias for visualize in all places (base.py, profile_visualize.py). Also added a couple of basic test cases similar to visualize just to check alias method works as expected. 

https://github.com/dask/dask/pull/6216

Do you suggest adding the alias note in documentation:
https://github.com/dask/dask/blob/7f5fc07f09c4626739c17fa9c46cdd78cc97224d/docs/source/graphviz.rst

**Comment by mindhash** [comment]
> I'm interested in doing this. Should i just create a method called visualise to alias visualize?

@1cybersheep1  sorry I didn't see your message as i started working on this.

**Comment by 1cybersheep1** [comment]
No problem! Good work!

**Comment by jcrist** [comment]
Fixed by #6216.
